cpp/include/cuml/experimental/fil/treelite_importer.hpp

```diff
@@ -331,6 +331,64 @@ struct treelite_importer {
     return result;
   }
 
+  // This function returns a modified copy of a given Treelite model if it contains
+  // at least one degenerate tree. If the model contains no degenerate tree, then
+  // the function returns nullptr.
+  auto convert_degenerate_trees(treelite::Model const& tl_model)
```

Comment:
Use explicit return types please for public functions. Trailing type is okay.

Reply:
Other public API functions in this header also use implicit return types. Should they be updated too?

Reply:
For internal APIs `auto` is fine but for public APIs explicit return types please. Other APIs should be updated as well but I will let you choose whether to do this in the diff of this PR or other follow on PRs that complete the promotion of experimental.

Reply:
I will make the change when we decide to move the header out of the experimental namespace.